packages/flutter/lib/src/rendering/stack.dart

```diff
@@ -520,8 +520,7 @@ class RenderStack extends RenderBox
     assert(_resolvedAlignment != null);
     bool hasNonPositionedChildren = false;
     if (childCount == 0) {
-      assert(constraints.biggest.isFinite);
-      return constraints.biggest;
+      return constraints.smallest;
```

Comment:
This is breaking existing behavior. If the biggest possible size the constraint allow is finite, the stack should continue to size itself to be as big as possible.

If that size is infinite, we could change this to the smallest possible size instead of asserting.